presto-main/src/main/java/com/facebook/presto/execution/executor/TaskExecutor.java

```diff
@@ -228,14 +230,21 @@ private synchronized void addRunnerThread()
         }
     }
 
-    public synchronized TaskHandle addTask(TaskId taskId, DoubleSupplier utilizationSupplier, int initialSplitConcurrency, Duration splitConcurrencyAdjustFrequency)
+    public synchronized TaskHandle addTask(
+            TaskId taskId,
+            DoubleSupplier utilizationSupplier,
+            int initialSplitConcurrency,
+            Duration splitConcurrencyAdjustFrequency,
+            OptionalInt maxDriversPerTask)
     {
         requireNonNull(taskId, "taskId is null");
         requireNonNull(utilizationSupplier, "utilizationSupplier is null");
+        checkArgument(!maxDriversPerTask.isPresent() || maxDriversPerTask.getAsInt() <= maximumNumberOfDriversPerTask,
+                "maxDriversPerTask must not be greater than the system default");
```

Comment:
`must not be` -> `cannot be`

Reply:
Also this may not be the "system default", it's the configured value (which can be different than the default value).

Reply:
Added a test case that sets the max drivers per task to 1 when the configured min and max is 2 and 4.